tests/serdetest.c

```diff
@@ -48,4 +49,213 @@ TEST (musig2, commitments_serialize_deserialize) {
+            assert(musig2_prepare_verifier(&aggr_pubkey, serialized_pubkey_list, NR_SIGNERS));
+            assert(secp256k1_xonly_pubkey_from_pubkey(ctx, &fuzz_aggr_pubkey, NULL, &mcs_list[0].mc.aggr_pubkey));
+            cmp_res = secp256k1_xonly_pubkey_cmp(ctx, &fuzz_aggr_pubkey, &aggr_pubkey);
+            ASSERT_NE(cmp_res, 0);
+            break;
+        default:
+            ASSERT_NE(err, MUSIG2_OK);
+    }
+}
+
+/* Fuzz a commitment then try to generate the aggregate R */
+TEST (musig2, fuzz_commitment_precomputation) {
+
+    secp256k1_context *ctx = secp256k1_context_create(SECP256K1_CONTEXT_VERIFY);
+    musig2_context_signer mcs_list[NR_SIGNERS]; // Array that holds NR_SIGNERS musig2_context_signer
+    musig2_context_signature mps[NR_SIGNERS];
+    unsigned char serialized_batch_list[NR_MESSAGES * NR_SIGNERS * V * MUSIG2_PUBKEY_BYTES_COMPRESSED];
+    unsigned char serialized_pubkey_list[NR_SIGNERS * MUSIG2_PUBKEY_BYTES_COMPRESSED];    // Signers' public key list
+    unsigned char signature[MUSIG2_BYTES];
+    secp256k1_pubkey pubkey;
+    MUSIG2_ERROR res;
+    int err;
+
+    res = musig2_helper_setup(mcs_list, serialized_pubkey_list, serialized_batch_list, NR_SIGNERS);
+    ASSERT_EQ(res, MUSIG2_OK);
+
+    int fuzz_index = (rand() % (NR_SIGNERS * V));
+    musig2_helper_fuzz_data(&serialized_batch_list[fuzz_index], MUSIG2_PUBKEY_BYTES_COMPRESSED);
+
+    res = musig2_helper_precomputation(serialized_pubkey_list, serialized_batch_list, mcs_list, NR_SIGNERS);
+
+    switch(res){
+        case MUSIG2_ERR_PARSE_COMM:
+            err = secp256k1_ec_pubkey_parse(ctx, &pubkey, &serialized_batch_list[fuzz_index], MUSIG2_PUBKEY_BYTES_COMPRESSED);
+            ASSERT_EQ(err, 0);
+            break;
+        case MUSIG2_ERR_AGGR_R:
+            err = secp256k1_ec_pubkey_parse(ctx, &pubkey, &serialized_batch_list[fuzz_index], MUSIG2_PUBKEY_BYTES_COMPRESSED);
+            ASSERT_EQ(err, 1);
+            break;
+        case MUSIG2_OK:
+            err = secp256k1_ec_pubkey_parse(ctx, &pubkey, &serialized_batch_list[fuzz_index], MUSIG2_PUBKEY_BYTES_COMPRESSED);
+            ASSERT_EQ(err, 1);
+
+            err = musig2_helper_sign(mcs_list, mps, NR_SIGNERS);
+            ASSERT_EQ(err, MUSIG2_OK);
+
+            err = musig2_aggregate_partial_sig(mps, signature, NR_SIGNERS);
+            ASSERT_EQ(err, MUSIG2_ERR_CMP_R);
+            break;
+        default:
+            err = secp256k1_ec_pubkey_parse(ctx, &pubkey, &serialized_batch_list[fuzz_index], MUSIG2_PUBKEY_BYTES_COMPRESSED);
+            ASSERT_EQ(err, 1);
+
+            err = musig2_helper_sign(mcs_list, mps, NR_SIGNERS);
+            ASSERT_EQ(err, MUSIG2_OK);
+
+            err = musig2_aggregate_partial_sig(mps, signature, NR_SIGNERS);
+            ASSERT_EQ(err, MUSIG2_OK);
+
+            err = musig2_helper_verify(serialized_pubkey_list, signature, MSG_1, MSG_1_LEN, NR_SIGNERS);
+            ASSERT_EQ(err, MUSIG2_INVALID);
```

Comment:
Also, same as above, in this test we only really care about serialisation. For the (in)validity of the signature, we already have another test, so no need to duplicate code/checks here:
```suggestion
        default:
            err = secp256k1_ec_pubkey_parse(ctx, &pubkey, &serialized_batch_list[fuzz_index], MUSIG2_PUBKEY_BYTES_COMPRESSED);
            ASSERT_EQ(err, 1);
```